### ISSUE TITLE
Use all space in toolbar for pads

### ISSUE THREAD

**Issue by Alisis33** [radimitrov/CSharpShellApp#199]
i written a feature request similar to this already, but pleas let me do it again.

I develope on a phone and a pad, but mostly on a pad.
on the phone there is not much space on the toolbar, so you spread the functions over the surface to keep them as accessible as possible. but on a pad, this is not needed. all functions can be staticaly on the toolbar. so why wasting all that empty space?

If all is directly an staticali accessable...
no switch on and off the copy/past
no arrow blocks the file menu. I can move directly a copy to a other file to insert it.
no reactivation of copy/past and cursor repositioning for multiple insertions (somethimes it not triggers....)
no serching for the search/replace in menu. (I sometimes click on code formating, i use a other formating style....)
no multiple clicks for undo/redo. nothing blocks the code
etc

pleas consider to place all main functionalitys (copy past undo redo search) permanent and static in the toolbar, if there is enough space for them. it would realy realy make coding easier and more fluid. if you don't like it, make it optional.
thanks.

**Comment by radimitrov** [comment]
Yes, that can actually work for a tablet.
And maybe phones in landscape orientation for the bluetooth keyboard users.

Search/replace. Paste, redo and undo if there is space too.
I might add the option to add any of the menu items to the keyboard quickbar.

You can change the "format code" code formatting style from the "curly brackets on new line" editor option.
However the project files browser panel being essentially constantly open and taking up code space isn't a good idea.

**Comment by Alisis33** [comment]
> However the project files browser panel being essentially constantly open and taking up code space isn't a good idea.
Yes, that was not me suggestion. The arrow is it which  is huding the button for it. I hafe the tools on top, and the arrow to close copy/past blocks the button for the flyout page.

**Comment by Alisis33** [comment]
A little of topic... (i can move it if you want)

I use this formating, basicaly compact and keywords from () separated, but not calls.

```
if (some) {
} else {
}

switch (some) {
case some:
break; case some:
break; }

Proc(par1, par2);
```

**Comment by radimitrov** [comment]
Hmm? I'm going to need a screenshot. Not sure how there is an arrow where I think you're saying.
And yes, that option will make the "Format code" menu item format it like you've shown.

**Comment by Alisis33** [comment]
If copy/past is active, there is a arrow pointing to left to close the copy/past. It blocks the 3line symbol to open the file manager. So more clicks again. If you put staticaly all symbols in the bar, the arrow is not neaded anymore.

**Comment by Alisis33** [comment]
![Screenshot_20221015-131828_C# Shell (C# Offline Compiler)](https://user-images.githubusercontent.com/103534710/195999576-54792bc5-3b77-46ea-9d55-a7a192d81093.jpg)

**Comment by Alisis33** [comment]
In case you update, pleas tell here, I would like to upgrade.

**Comment by radimitrov** [comment]
There is actually much less space usable than is visible. The white parts of the picture illustrate it approximately. That woudn't fit on most devices. And forcing it to fit would require the other buttons to become smaller, which isn't very good.
I'll make the arrow unnecessary when the option to put the selection buttons in the bottom quickbar is enabled.
![toolbar_free_space](https://user-images.githubusercontent.com/56889141/196036092-638d8806-d86f-4915-be65-83019dba8fe5.jpg)

**Comment by Alisis33** [comment]
My suggestion it to simply use all the space in larger devices like pads, there is then no such problems, all icons could be staticaly there and no closing button is needed. And leave the mechanism like it is for smal devices if there is no space, or move it beside the file flyoutpage button, if there is space.

I would maybe not close automatically the copy/past bar after a past, so you can past multiple times. Now you hafe allways to reactivate the bar. (Sometimes you have to click around because it not trigger)

But like I sed, I work on pad mostly, and there it would be great to hafe all functions allways available.

**Comment by Alisis33** [comment]
And, yes, don't make the buttons smaller.
On a pad you could make the bar even a little bit higher.

**Comment by radimitrov** [comment]
Ok, I've added debug/step, search, redo, undo and format code as toolbar icons. 
They should show when there is enough space. An 8 inch tablet should be enough in portrait mode.

**Comment by Alisis33** [comment]
Tanks much, this definitly reduce some neurological stress.

But in v 6.12.0 it not shows copy/paste in the toolbar. Instead debug-compile and code formating.
I woult take out code formatting and then debug-compile if there is not enough space. Copy/paste us used way more frequently. In case of smal space, I would efen sacrifice search for copy/paste/cut.

**Comment by Alisis33** [comment]
It is copy paste I miss the most. I use it so often that is should be permanently acessable with one click.
(To be honest, it is wath causes the most frustration now, it is what costs the most extra clicks. To code on android pads and phones is not the same as on PC with hardware keyboard and ctrl.)

Cold it be optional?
So every user makes the tool best fit for his needs.

**Comment by radimitrov** [comment]
I'll add it as an option

**Comment by Alisis33** [comment]
Not sure if I am blind and overlooked the option, but I can not find a change.
Copy paste is not static, only
Debug,search,undo,redo,format,compile
V 2.6.53

My sugestion was
Select,Cut,copy,paste,undo,redo,compile
And for what ever more there is space.

**Comment by radimitrov** [comment]
I haven't released an update yet. The option will be in the editor settings.

**Comment by Alisis33** [comment]
Ohh i was a bit to fast, the Version now hafe it.
Tanks much!
Always nice to see if developers have ear for the users. Specialy we are here all developers too, sinking down in feature requests and todo lists... ;-)

**Comment by Alisis33** [comment]
Can be closed
